sys/linux/sys.txt

```diff
@@ -494,6 +495,9 @@ pidfd_send_signal(fd fd_pidfd, sig signalno, info ptr[in, siginfo], flags const[
 pidfd_open(pid pid, flags const[0]) fd_pidfd
 pidfd_getfd(pidfd fd_pidfd, fd fd, flags const[0]) fd
 
+close_range(fd fd, max_fd fd, flags const[CLOSE_RANGE_UNSHARE])
+close_range$MAXFD(fd fd, max_fd const[INT_MAX], flags const[CLOSE_RANGE_UNSHARE])
```

Comment:
I'm not sure how helpful this special case is.  IIRC syzkaller has heuristics to try random integers occasionally, so we should probably just rely on those for this corner case.

Reply:
Yes, it should try this without special descriptions.

Reply:
The reason I had this special case is because according to the documentation. "INT_MAX should be passed in most of the time", and I feel like trying random integers occasionally wouldn't give us that frequency we desire. Should I just delete this special case?

Reply:
Is it accepted as a signed or unsigned integer in the kernel? syzkaller will pass -1 for fd from time to time, and if we declare it as `fd[opt]` it will pass -1 more. And if it's declared as unsigned, then -1 will turn into a very large int. Which can do what we want.

I would either delete this, or use a union to specialize this particular arg rather than the whole syscall:
```
fd_or_int_max [
	fd	fd
	max	const[INT_MAX, int32]
]
```

I would assume that syzkaller will be able to test all possible combinations (closing some/none/all/etc fds) without any particular issues without this specialization b/c this is quite local to this single syscall and just requires passing a particular value in a single argument. It's harder to test cases where multiples values need to be in particular values (exponential explosion).
But this is not exact science. There is no single right answer to everything.

Reply:
I think we should delete it.  All it does is increase the frequency we pass `INT_MAX`, which it's hard to say whether more bugs or coverage will be found or not.  Let's keep things simpler unless we have evidence that the frequency needs to be adjusted.

Reply:
If Matt also says this, let's go for the simpler descriptions.

Reply:
I see. Thanks for the explanation, I've just changed this!